swap/src/cli/cancel_and_refund.rs

```diff
@@ -0,0 +1,116 @@
+use crate::bitcoin::wallet::Subscription;
+use crate::bitcoin::{parse_rpc_error_code, RpcErrorCode, Wallet};
+use crate::protocol::bob::BobState;
+use crate::protocol::Database;
+use anyhow::{bail, Result};
+use bitcoin::Txid;
+use std::sync::Arc;
+use uuid::Uuid;
+
+pub async fn cancel_and_refund(
+    swap_id: Uuid,
+    bitcoin_wallet: Arc<Wallet>,
+    db: Arc<dyn Database>,
+) -> Result<BobState> {
+    match cancel(swap_id, bitcoin_wallet.clone(), db.clone()).await {
+        Ok((_, subscription, _)) => subscription.wait_until_final().await?,
```

Comment:
Do we need to wait for the cancel transaction to be confirmed? I don't really see any reason to do so.

Reply:
Yes the refund and punish transactions spend the output of the cancel transaction, so I believe it would at least have to be confirmed. 

“ Alice constructs and shares with Both an adaptor signature on Tx_refund spending from the output of Tx_cancel to address_b.” 

https://comit.network/blog/2020/10/06/monero-bitcoin/

Reply:
But if I'm not mistaken, Bitcoin allows you to publish transactions that spend unconfirmed outputs. They (Refund tx) just won't be confirmed until their parent (Cancel Tx) has been included in a block.

Not waiting for the Cancel Tx to be confirmed allows both TxCancel and TxRefund to be included in the same block.

Reply:
Done